### PR TITLE
Feature/#443 add map pool assignment

### LINKED ISSUE #443: Allow reading and writing of map params in db v109

### PR DESCRIPTION
**Pull request by Sheikah45**
Closes #443

### CODE REVIEW

File: src/inttest/java/com/faforever/api/data/MapPoolElideTest.java [hunk 1]
```diff
@@ -16,12 +16,34 @@
...
+          "type": "mapVersion",
+          "id": "1"
+        }
+      }
+    }
+  }
+}
+   */
+
+  private static final String NEW_LADDER_MAP_BODY = "{\"data\":{\"type\":\"mapPoolAssignment\",\"relationships\":{\"mapPool\":{\"data\":{\"type\":\"mapPool\",\"id\":\"1\"}},\"mapVersion\":{\"data\":{\"type\":\"mapVersion\",\"id\":\"1\"}}}}}";
```
Comment (Brutus5000) [inline src/inttest/java/com/faforever/api/data/MapPoolElideTest.java]: Use multiline String in the variable instead

File: src/main/java/com/faforever/api/data/domain/MapPoolAssignment.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+@Include(rootLevel = true, type = "mapPoolAssignment")
+@CreatePermission(expression = WriteMatchmakerMapCheck.EXPRESSION)
+@UpdatePermission(expression = WriteMatchmakerMapCheck.EXPRESSION)
+@DeletePermission(expression = WriteMatchmakerMapCheck.EXPRESSION)
+public class MapPoolAssignment {
+  private int id;
+  private MapPool mapPool;
+  private MapVersion mapVersion;
+  private int weight;
+  private String mapParams;
```
Comment (Brutus5000) [inline src/main/java/com/faforever/api/data/domain/MapPoolAssignment.java:35]: Do we ever care about the contents? Then maybe Map<String, Object> or something like JsonObject

**Review by Brutus5000 — CHANGES_REQUESTED** [review]

File: src/main/java/com/faforever/api/data/domain/MapPoolAssignment.java (hunk 2 above)
Comment (Sheikah45) [reply src/main/java/com/faforever/api/data/domain/MapPoolAssignment.java:35]: Not sure what you mean here. When I try to use those types I just get that it doesn't know how to handle them.

**Comment by Sheikah45** [comment]
I went ahead and made this rely on version v112 of the database so checks need to be rerun once that is released

File: src/main/java/com/faforever/api/data/domain/MapPoolAssignment.java (hunk 2 above)
Comment (Brutus5000) [reply src/main/java/com/faforever/api/data/domain/MapPoolAssignment.java:35]: Ah yes, Hibernate doesn't support this. Shame. Jackson can handle this.

**Review by Brutus5000 — APPROVED** [review]

**Pull request merged**